docs/pages/_app.js

```diff
@@ -222,6 +223,23 @@ function AppWrapper(props) {
             <PageContext.Provider value={{ activePage, pages: productPages }}>
               <ThemeProvider>
                 <DocsStyledEngineProvider cacheLtr={emotionCache}>
+                  {asPathWithoutLang !== '/' && (
+                    <GlobalStyles
+                      styles={(theme) => ({
+                        html: {
+                          // to override CssBaseline in BrandingCssVarsProvider when navigating between pages
+                          colorScheme: `${theme.palette.mode} !important`,
+                          // If the user lands on/refreash homepage, the CssBaseline with CSS variables are created inline in the html due to server-side generation
+                          // When user navigates to other pages, the CssBaseline in those pages can't win because the css-global is injected above server-side generation
+                          // So these variables are needed to make the body background and color render correctly.
+                          '--muidocs-palette-background-default': (theme.vars || theme).palette
+                            .background.default,
+                          '--muidocs-palette-text-primary': (theme.vars || theme).palette.text
+                            .primary,
+                        },
+                      })}
+                    />
+                  )}
```

Comment:
Are we sure about this part? It sounds like there are styles leakings that should be automatically unmounted by aren't. What caught my attention is the !important and the fact that we are modifying the _app.js

Reply:
Found a better way so this part is removed:

```js
colorScheme: `${theme.palette.mode} !important`,
```